### PR TITLE
MDI support for LAMMPS + LATTE (tight-binding code)

### PR DESCRIPTION
**Pull request by sjplimp**
**Summary*

Some adjustments & bug fixes to MDI support in LAMMPS to make it work with the LATTE QM code.  Added an examples/QM
dir with a LATTE sub-dir for example use cases.

**Related Issue(s)**

N/A

**Author(s)**

Steve with help from Christian Negre (LANL, LATTE developer)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

**Comment by sjplimp** [comment]
@akohlmey Still some testing we need to do on this for a plugin issue - not sure whether it is a LATTE or MDI or LAMMPS issue.
Now Taylor and Christian have access to this version.

File: src/MDI/mdi_plugin.cpp [hunk 1]
```diff
@@ -74,18 +74,23 @@ MDIPlugin::MDIPlugin(LAMMPS *_lmp, int narg, char **arg) : Pointers(_lmp)
...
   // error checks
 
-  if (!mdi_arg || !infile_arg || !lammps_command)
-    error->all(FLERR, "MDI plugin must specify mdi, infile, command keywords");
+  if (!mdi_arg || !lammps_command)
+    error->all(FLERR, "MDI plugin must specify mdi and command keywords");
 
   // build full plugin_args string for args to plugin library
 
-  int n = strlen(mdi_arg) + strlen(infile_arg) + strlen(extra_arg) + 16;
```
Comment (akohlmey) [inline src/MDI/mdi_plugin.cpp:80]: This kind of thing is much cleaner and simpler done by assembling the command line string(s) in a std::string variable using `fmt::format()` and then using utils::strdup to make a dynamically allocated copy.

**Review by akohlmey — APPROVED** [review]

**Comment by akohlmey** [comment]
@sjplimp did some final cosmetic updates with clang-format. Unless there is an objection, I will merge this as soon as the pending tests clear, i.e. in about 30-60 minutes.

**Comment by sjplimp** [comment]
good to go from my perspective - thanks

**Pull request merged**